### PR TITLE
Fix code smell for widget folder

### LINKED ISSUE #520: Better for you to follow the stile guide: Client (widget folder)
## Description:
For the long time we didn't following the style, namely chapter _Names_ rule 2. It's time to change that!
This should remove such warnings:
```log
/home/runner/work/Juniorgram/Juniorgram/Network/Public/Include/Network/Primitives.hpp: In constructor ‘Network::ChannelInfo::ChannelInfo(uint64_t, uint64_t, const string&)’:
/home/runner/work/Juniorgram/Juniorgram/Network/Public/Include/Network/Primitives.hpp:32:9: warning: declaration of ‘channelName’ shadows a member of ‘Network::ChannelInfo’ [-Wshadow]
   32 |         : creatorID(creatorID), channelID(channelID), channelName(channelName)
      |     
```

## Acceptance Criteria:
Follow the rule and rename variables in proper way with underscore.

### PR DESCRIPTION
**Pull request by Agrifon**

### CODE REVIEW

File: Client/Client.Qt/Widgets/LoadingSpinner.hpp [hunk 1]
```diff
@@ -47,7 +47,6 @@ private slots:
     /// Put on center of page
     void updatePosition();
 
-private:
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/LoadingSpinner.hpp:50]: This private separate private methods and private fields

File: Client/Client.Qt/Widgets/ChatWidget.hpp [hunk 2]
```diff
@@ -18,7 +18,7 @@ class ChatWidget : public QWidget
      * @brief Constructor for chat widget
      * @param parent Parent widget
      */
-    explicit ChatWidget(std::string channelName, QWidget* parent = nullptr);
+    explicit ChatWidget(std::string& channelName, QWidget* parent = nullptr);
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/ChatWidget.hpp]: Should be std::string_view

File: Client/Client.Qt/Widgets/ChatHistory.cpp [hunk 3]
```diff
@@ -261,7 +261,7 @@ std::pair<int, int> ChatHistory::findVisible() const
     }
     int middle = (right + left) / 2;
     int index  = middle;
-    int size   = int(_messageList.size());
+    auto size   = int(_messageList.size());
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/ChatHistory.cpp]: c-style casting should be changed

File: Client/Client.Qt/Widgets/AbstractButton.hpp [hunk 4]
```diff
@@ -36,7 +36,7 @@ class AbstractButton : public QWidget
      * @brief Enable/disable this button
      * @param disabled Enable/disable this button
      */
-    void setDisabled(bool disabled = true);
+    void setDisabledButton(bool disabled = true);
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/AbstractButton.hpp]: Rename to "setButtonDisabled"

File: .github/workflows/GCC Linux Build Sonar.yml [hunk 5]
```diff
@@ -16,15 +16,29 @@ jobs:
       BUILD_WRAPPER_OUT_DIR: /home/runner/.build_wrapper_out/
     strategy:
       matrix:
-        BUILD_TYPE: [Release] 
+        BUILD_TYPE: [Debug]
+    
+    services:
+      postgres:
+        image: postgres
+        env:
```
Comment (Orange-hanter) [inline .github/workflows/GCC Linux Build Sonar.yml]: Hmmm....

File: Client/Client.Qt/Widgets/AbstractMessageWidget.hpp [hunk 6]
```diff
@@ -33,10 +33,10 @@ class AbstractMessageWidget : public QWidget
 
 protected:
     // Getters and setters for derived classes
-    uint64_t getUserID()             const { return _userID; };
-    uint64_t getMessageID()          const { return _messageID; }
-    const QString&  getMessageText() const { return _messageText; }
-    const QString&  getUserName()    const { return _userName; }
+    uint64_t getUserID()                   const { return _userID; };
+    virtual uint64_t getMessageID()        const { return _messageID; }
```
Comment (ivanboika) [inline Client/Client.Qt/Widgets/AbstractMessageWidget.hpp]: Why did you make only this method virtual? Was it code smell?

File: Client/Client.Qt/Widgets/ChatHistory.cpp [hunk 7]
```diff
@@ -62,7 +61,7 @@ void ChatHistory::addMessage(const Network::MessageInfo& messageInfo)
     {
         if (copy->reactions != messageInfo.reactions)
         {
-            auto widgetCopy = std::find_if(_messageList.begin(), _messageList.end(), [messageInfo](const std::unique_ptr<MessageWidget>& mw)
+            auto widgetCopy = std::find_if(_messageList.begin(), _messageList.end(), [messageInfo](const std::shared_ptr<MessageWidget>& mw)
```
Comment (ivanboika) [inline Client/Client.Qt/Widgets/ChatHistory.cpp:64]: MessageList field is unique_ptr, what's the reason to change it on shared even if it's const? Again, if this is sonar code smell I don't mind

File: Client/Client.Qt/Widgets/ChatHistory.cpp [hunk 8]
```diff
@@ -306,3 +286,26 @@ std::pair<int, int> ChatHistory::findVisible() const
...
+        if (middle < top) left = (right + left) / 2;
+        if (middle > top) right = (right + left) / 2;
+        it     = _replyList.find(int(_messageList[(right - left) / 2]->getMessageID()));
+        middle = _messageList[(right - left) / 2]->pos().y();
+        if (it != _replyList.end())
+        {
+            middle = it->second->pos().y();
+        }
+    }
+}
```
Comment (ivanboika) [inline Client/Client.Qt/Widgets/ChatHistory.cpp]: Add empty line there please

File: Client/Client.Qt/Widgets/InputFields.cpp [hunk 9]
```diff
@@ -26,49 +33,43 @@ class InputStyle : public QCommonStyle
...
 
 template <typename InputClass>
-InputStyle<InputClass>* InputStyle<InputClass>::_instance = nullptr;
+std::shared_ptr<InputStyle<InputClass>> InputStyle<InputClass>::_instance = nullptr;
 
 FlatInput::FlatInput(QWidget* parent, const QString& placeholder, bool password) : QLineEdit(parent)
 {
     setFont(st::defaultFont);
-    auto* regexpvalidator = new QRegularExpressionValidator;
+    auto regexpvalidator = std::make_shared<QRegularExpressionValidator>();
```
Comment (ivanboika) [inline Client/Client.Qt/Widgets/InputFields.cpp]: Where is camelCase?

File: Client/Client.Qt/Widgets/ScrollArea.cpp [hunk 10]
```diff
@@ -33,28 +33,34 @@ void ScrollBar::updateBar(bool)
...
+        int h  = sh ? int32_t((rh * int64_t(area()->height())) / sh) : 0;
         if (h >= rh || !area()->scrollTopMax() || rh < 30)
         {
             if (!isHidden()) hide();
             return;
         }
 
         if (h <= 30) h = 30;
-        int stm = area()->scrollTopMax(), y = stm ? int32_t(((rh - h) * int64_t(area()->scrollTop())) / stm) : 0;
+        int stm = area()->scrollTopMax();
```
Comment (ivanboika) [inline Client/Client.Qt/Widgets/ScrollArea.cpp]: As for me, it's difficult to understand the meaning of this variables

File: Client/Client.Qt/Widgets/AbstractMessageWidget.hpp (hunk 6 above)
Comment (Agrifon) [reply Client/Client.Qt/Widgets/AbstractMessageWidget.hpp]: Yes, it was code smell

File: Client/Client.Qt/Widgets/ScrollArea.cpp [hunk 11]
```diff
@@ -101,13 +107,15 @@ void ScrollBar::paintEvent(QPaintEvent*)
     if (opacity == 0.) return;
 
     QPainter p(this);
-    auto     deltal = _vertical ? _st->deltax : 0, deltar = _vertical ? _st->deltax : 0;
-    auto     deltat = _vertical ? 0 : _st->deltax, deltab = _vertical ? 0 : _st->deltax;
+    auto deltal = _vertical ? _st->deltax : 0;
```
Comment (ivanboika) [inline Client/Client.Qt/Widgets/ScrollArea.cpp]: I guess, there should be underscore, ofc if that's not one word

File: Client/Client.Qt/Widgets/ScrollArea.cpp [hunk 12]
```diff
@@ -252,11 +260,13 @@ void ScrollBar::mousePressEvent(QMouseEvent* e)
     }
     else
     {
-        int32_t val = _vertical ? e->pos().y() : e->pos().x(), div = _vertical ? height() : width();
+        int32_t val = _vertical ? e->pos().y() : e->pos().x();
+        int32_t div = _vertical ? height() : width();
```
Comment (ivanboika) [inline Client/Client.Qt/Widgets/ScrollArea.cpp]: image
This name is already used as a visual studio method

File: Client/Client.Qt/Widgets/TextEdit.hpp [hunk 13]
```diff
@@ -47,16 +47,16 @@ public slots:
...
-    void styleButtonClick(const QString& symbolStart, const QString& symbolEnd);
+    void styleButtonClick(const QString& symbolStart, const QString& symbolEnd) const;
 
 private:
-    void delSymbolsInSelection(QString& text, int& start, int& end, int symbolSize);
-    void delSymbolsOutSelection(QString& text, int& start, int& end, int symbolSize);
-    void insertSymbolsInSelection(QTextCursor& cursor, int& start, int& end, int symbolSize, const QString& symbolStart,
-                                  const QString& symbolEnd);
-    void selectText(QTextCursor& cursor, int start, int end);
+    void delSymbolsInSelection(QString& text, int start, int end, int symbolSize) const;
```
Comment (ivanboika) [inline Client/Client.Qt/Widgets/TextEdit.hpp:53]: Should be tested I suppose, because the author of this code made '&' with int with a reason, maybe

**Review by ivanboika — CHANGES_REQUESTED** [review]
Some fixing needed

File: Client/Client.Qt/Widgets/ScrollArea.cpp (hunk 12 above)
Comment (ivanboika) [reply Client/Client.Qt/Widgets/ScrollArea.cpp]: Only this left

File: Client/Client.Qt/Widgets/ScrollArea.cpp [hunk 14]
```diff
@@ -33,39 +33,46 @@ void ScrollBar::updateBar(bool)
...
         _scrollMax = newMax;
         area()->rangeChanged(oldMax, newMax, _vertical);
     }
     if (_vertical)
     {
-        int sh = area()->scrollHeight(), rh = height(), h = sh ? int32_t((rh * int64_t(area()->height())) / sh) : 0;
-        if (h >= rh || !area()->scrollTopMax() || rh < 30)
+        int scrollHeight = area()->scrollHeight();
+        int rHeight = height();
+        int height       = scrollHeight ? int32_t((rHeight * int64_t(area()->height())) / scrollHeight) : 0;
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/ScrollArea.cpp]: Here functionality could be moved to inline function

File: Client/Client.Qt/Widgets/ScrollArea.cpp [hunk 15]
```diff
@@ -33,39 +33,46 @@ void ScrollBar::updateBar(bool)
...
+        if (height >= rHeight || !area()->scrollTopMax() || rHeight < 30)
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/ScrollArea.cpp]: What is rHeight?

File: Client/Client.Qt/Widgets/ScrollArea.cpp [hunk 16]
```diff
@@ -33,39 +33,46 @@ void ScrollBar::updateBar(bool)
...
             if (!isHidden()) hide();
             return;
         }
 
-        if (h <= 30) h = 30;
-        int stm = area()->scrollTopMax(), y = stm ? int32_t(((rh - h) * int64_t(area()->scrollTop())) / stm) : 0;
-        if (y > rh - h) y = rh - h;
+        if (height <= 30) height = 30;
+        int scrollTopMax = area()->scrollTopMax();
+        int left            = scrollTopMax ? int32_t(((rHeight - height) * int64_t(area()->scrollTop())) / scrollTopMax) : 0;
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/ScrollArea.cpp]: Also, what is left? Left corner or left button...

Comment (Orange-hanter) [reply Client/Client.Qt/Widgets/ScrollArea.cpp]: Same as above. Move it to the function

File: Client/Client.Qt/Widgets/ScrollArea.cpp [hunk 17]
```diff
@@ -33,39 +33,46 @@ void ScrollBar::updateBar(bool)
...
+        if (height >= rHeight || !area()->scrollTopMax() || rHeight < 30)
         {
             if (!isHidden()) hide();
             return;
         }
 
-        if (h <= 30) h = 30;
-        int stm = area()->scrollTopMax(), y = stm ? int32_t(((rh - h) * int64_t(area()->scrollTop())) / stm) : 0;
-        if (y > rh - h) y = rh - h;
+        if (height <= 30) height = 30;
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/ScrollArea.cpp]: Magic constants...

File: Client/Client.Qt/Widgets/ScrollArea.cpp [hunk 18]
```diff
@@ -229,8 +238,8 @@ void ScrollBar::mouseMoveEvent(QMouseEvent* e)
     setOverBar(_bar.contains(e->pos()));
     if (_moving)
     {
-        int delta = 0, barDelta = _vertical ? (area()->height() - _bar.height()) : (area()->width() - _bar.width());
-        if (barDelta > 0)
+        int delta = 0;
+        if (int barDelta = _vertical ? (area()->height() - _bar.height()) : (area()->width() - _bar.width()); barDelta > 0)
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/ScrollArea.cpp]: bad practice to put long expresion in if

File: Client/Client.Qt/Widgets/ScrollArea.cpp (hunk 12 above)
Comment (Orange-hanter) [reply Client/Client.Qt/Widgets/ScrollArea.cpp]: Div is not clear name, at least by this reason it should be renamed.
Try _proportion_.

**Review by Orange-hanter — CHANGES_REQUESTED** [review]
Files reviewed:
ScrollArea
TextEdit
TitleWidget

**Review by ivanboika — COMMENTED** [review]
One fuzzy moment with naming, after create ticket of 'menu button' bug
And rebase please

**Review by ivanboika — APPROVED** [review]

**Review by memorisecodead — APPROVED** [review]

File: Client/Client.Qt/Widgets/AbstractMessageWidget.hpp (hunk 6 above)
Comment (Orange-hanter) [reply Client/Client.Qt/Widgets/AbstractMessageWidget.hpp]: Strange situation. Virtual and protected...
Check please do we really need this combination?

**Review by memorisecodead — APPROVED** [review]

File: Client/Client.Qt/Widgets/ChatHistory.cpp [hunk 19]
```diff
@@ -76,37 +75,37 @@ void ChatHistory::addMessage(const Models::MessageInfo& messageInfo)
...
     msg->resize(history->width() - 25, msg->expectedHeight());
     history->setMinimumHeight(history->minimumHeight() + msg->expectedHeight() + 10);
 
-    _messageList.push_back(std::unique_ptr<MessageWidget>(msg));
+    _messageList.push_back(msg);
     _messages.push_back(messageInfo);
 
-    std::sort(_messageList.begin(), _messageList.end(), [](const std::unique_ptr<MessageWidget>& lhs, const std::unique_ptr<MessageWidget>& rhs)
+    std::sort(_messageList.begin(), _messageList.end(), [](const std::shared_ptr<MessageWidget>& lhs, const std::shared_ptr<MessageWidget>& rhs)
         { return *lhs < *rhs;  });
```
Comment (slavexx) [inline Client/Client.Qt/Widgets/ChatHistory.cpp:90]: The same question about changed pointer

File: Client/Client.Qt/Widgets/ChatHistory.cpp [hunk 20]
```diff
@@ -62,7 +61,7 @@ void ChatHistory::addMessage(const Models::MessageInfo& messageInfo)
     {
         if (copy->_reactions != messageInfo._reactions)
         {
-            auto widgetCopy = std::find_if(_messageList.begin(), _messageList.end(), [messageInfo](const std::unique_ptr<MessageWidget>& mw)
+            auto widgetCopy = std::find_if(_messageList.begin(), _messageList.end(), [messageInfo](const std::shared_ptr<MessageWidget>& mw)
```
Comment (slavexx) [inline Client/Client.Qt/Widgets/ChatHistory.cpp:64]: Why is the pointer changed there? There wasn't code smell.

File: Client/Client.Qt/Widgets/ChatHistory.hpp [hunk 21]
```diff
@@ -74,20 +74,21 @@ private Q_SLOTS:
...
-private:
     bool                                                   _alreadyScrolling = false;
     std::int32_t                                           _left             = -1;
     std::unique_ptr<ScrollArea>                            _scrollArea;
-    std::map<int32_t, std::unique_ptr<ReplyMessageWidget>> _replyList;
-    std::vector<std::unique_ptr<MessageWidget>>            _messageList;
-    std::vector<Models::MessageInfo>                       _messages;
-    std::vector<Models::ReplyInfo>                         _replies;
+    std::map<int32_t, std::shared_ptr<ReplyMessageWidget>> _replyList;
+    std::vector<std::shared_ptr<MessageWidget>>            _messageList;
```
Comment (slavexx) [inline Client/Client.Qt/Widgets/ChatHistory.hpp:89]: The same question about changed pointer

File: Client/Client.Qt/Widgets/ScrollArea.cpp [hunk 22]
```diff
@@ -33,39 +33,52 @@ void ScrollBar::updateBar(bool)
...
-        int stm = area()->scrollTopMax(), y = stm ? int32_t(((rh - h) * int64_t(area()->scrollTop())) / stm) : 0;
-        if (y > rh - h) y = rh - h;
+        if (lHeight <= minHeight) lHeight = minHeight;
+        int scrollTopMax = area()->scrollTopMax();
+        int scrollTop    = area()->scrollTop();
+        int leftCorner   = getLeftCorner(scrollTopMax, canvasHeight, lHeight, scrollTop);
+        if (leftCorner > canvasHeight - lHeight) leftCorner = canvasHeight - lHeight;
 
-        newBar = QRect(_st->deltax, y, width() - 2 * _st->deltax, h);
+        newBar = QRect(_st->deltax, leftCorner, width() - 2 * _st->deltax, lHeight);
```
Comment (slavexx) [inline Client/Client.Qt/Widgets/ScrollArea.cpp]: I think, name "delta_x" is better (as you did below in line 130)

File: Client/Client.Qt/Widgets/ScrollArea.cpp [hunk 23]
```diff
@@ -33,39 +33,52 @@ void ScrollBar::updateBar(bool)
...
-        int slm = area()->scrollLeftMax(), x = slm ? int32_t(((rw - w) * int64_t(area()->scrollLeft())) / slm) : 0;
-        if (x > rw - w) x = rw - w;
+        if (lWidth <= minWidth) lWidth = minWidth;
+        int scrollLeftMax = area()->scrollLeftMax();
+        int scrollLeft    = area()->scrollLeft();
+        int leftCorner    = getLeftCorner(scrollLeftMax, canvasWidth, lWidth, scrollLeft);
+        if (leftCorner > canvasWidth - lWidth) leftCorner = canvasWidth - lWidth;
 
-        newBar = QRect(x, _st->deltax, w, height() - 2 * _st->deltax);
+        newBar = QRect(leftCorner, _st->deltax, lWidth, height() - 2 * _st->deltax);
```
Comment (slavexx) [inline Client/Client.Qt/Widgets/ScrollArea.cpp]: Suggestion to change to "delta_x"

File: Client/Client.Qt/Widgets/LoadingSpinner.hpp [hunk 24]
```diff
@@ -47,10 +47,10 @@ private slots:
...
+    //variable field
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/LoadingSpinner.hpp]: Is that important to leave this comment?

**Review by slavexx — APPROVED** [review]

File: Client/Client.Qt/Widgets/Menu.hpp [hunk 25]
```diff
@@ -84,7 +84,7 @@ class Menu : public QWidget
     void handleMousePress(QPoint globalPosition);
 
     void setMouseReleaseDelegate(std::function<void(QPoint globalPosition)> delegate) { _mouseReleaseDelegate = std::move(delegate); }
-    void handleMouseRelease(QPoint globalPosition);
+    void handleMouseRelease(QPoint globalPosition) const;
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/Menu.hpp:86]: Look at the start of fle. Compare .cpp and .hpp files.

File: Client/Client.Qt/Widgets/MessageWidget.cpp [hunk 26]
```diff
@@ -13,41 +13,33 @@
 
 MessageWidget::MessageWidget(QWidget* history, QString message, uint64_t userId, uint64_t messageId, qint64 utc, QString username,
                              const Style::MessageWidget& st)
-    : AbstractMessageWidget(history, message, messageId, username, userId, st) 
-    , _datetime(QDateTime::fromMSecsSinceEpoch(utc))
-{   
+    : AbstractMessageWidget(history, message, messageId, username, userId, st), _datetime(QDateTime::fromMSecsSinceEpoch(utc))
+{
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/MessageWidget.cpp:14]: Include dublication. Compare both files

File: Client/Client.Qt/Widgets/Page.cpp [hunk 27]
```diff
@@ -10,9 +10,9 @@ Page::Page(QWidget* parent) : QFrame(parent)
     connect(ReceiverManager::instance(), &ReceiverManager::onDisconnect, this, &Page::onResume);
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/Page.cpp:10]: Change the todo format to doxygen compatible

File: Client/Client.Qt/Widgets/TextEdit.cpp [hunk 28]
```diff
@@ -96,33 +96,34 @@ void TextEdit::styleButtonClick(const QString& symbolStart, const QString& symbo
     }
 }
 
-void TextEdit::delSymbolsInSelection(QString& text, int& start, int& end, int symbolSize)
+void TextEdit::delSymbolsInSelection(QString& text, int start, int end, int symbolSize) const
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/TextEdit.cpp:99]: Explain next time what happend here?

**Review by Orange-hanter — CHANGES_REQUESTED** [review]

File: Client/Client.Qt/Widgets/LoadingSpinner.hpp [hunk 29]
```diff
@@ -47,10 +47,9 @@ private slots:
     /// Put on center of page
     void updatePosition();
 
-private:
     const Style::LoadingSpinner& _st;
     std::unique_ptr<QTimer>      _timer;
     bool                         _disableParentWhenSpinning;
-    int32_t                      _currentCounter;
-    bool                         _isSpinning;
+    int32_t                      _currentCounter = 0;
```
Comment (memorisecodead) [inline Client/Client.Qt/Widgets/LoadingSpinner.hpp]: I think so there don't use int32_t. For this reason enough uint16_t.

image

**Review by memorisecodead — APPROVED** [review]

File: Client/Client.Qt/Widgets/Page.cpp (hunk 27 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/Page.cpp:10]: @Orange-hanter, @ivanboika I did.

File: Client/Client.Qt/Widgets/LoadingSpinner.hpp (hunk 24 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/LoadingSpinner.hpp]: @Orange-hanter, @ivanboika No

File: Client/Client.Qt/Widgets/AbstractMessageWidget.hpp (hunk 6 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/AbstractMessageWidget.hpp]: @Orange-hanter, @ivanboika I did. We don't need this combination.

File: Client/Client.Qt/Widgets/Menu.hpp (hunk 25 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/Menu.hpp:86]: @Orange-hanter , @ivanboika  I did.

File: Client/Client.Qt/Widgets/MessageWidget.cpp (hunk 26 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/MessageWidget.cpp:14]: @Orange-hanter , @ivanboika  I did.

File: Client/Client.Qt/Widgets/LoadingSpinner.hpp [hunk 30]
```diff
@@ -47,10 +47,9 @@ private slots:
...
     void updatePosition();
 
-private:
     const Style::LoadingSpinner& _st;
     std::unique_ptr<QTimer>      _timer;
     bool                         _disableParentWhenSpinning;
-    int32_t                      _currentCounter;
-    bool                         _isSpinning;
+    uint16_t                     _currentCounter = 0;
+    bool                         _isSpinning = false;
```
Comment (slavexx) [inline Client/Client.Qt/Widgets/LoadingSpinner.hpp]: I think, equal signs should be set according code styling (like in ch. "Formatting", p.9)

**Review by slavexx — APPROVED** [review]
Minor changes are needed

File: Client/Client.Qt/Widgets/LoadingSpinner.hpp (hunk 24 above)
Comment (Orange-hanter) [reply Client/Client.Qt/Widgets/LoadingSpinner.hpp]: So delete them

File: Client/Client.Qt/Widgets/Menu.hpp (hunk 25 above)
Comment (Orange-hanter) [reply Client/Client.Qt/Widgets/Menu.hpp:86]: I see the same include in both files. Why?

File: Client/Client.Qt/Widgets/MessageWidget.cpp (hunk 26 above)
Comment (Orange-hanter) [reply Client/Client.Qt/Widgets/MessageWidget.cpp:14]: _#Include_ is still duplicated here

File: Client/Client.Qt/Widgets/TextEdit.cpp [hunk 31]
```diff
@@ -36,22 +36,22 @@ TextEdit::TextEdit(QWidget* parent) : QWidget(parent), _settings(Settings::getIn
 
     setLayout(_mainVerticalLayout.get());
 
-    _boldnessButton->setClickCallback([&]() { styleButtonClick(_boldSymbolOpen, _boldSymbolClose); });
-    _italicButton->setClickCallback([&]() { styleButtonClick(_italicSymbolOpen, _italicSymbolClose); });
-    _underlineButton->setClickCallback([&]() { styleButtonClick(_underlineSymbolOpen, _underlineSymbolClose); });
-    _sendButton->setClickCallback([&]() { sendButtonClick(); });
+    _boldnessButton->setClickCallback([this]() { this->styleButtonClick(this->_boldSymbolOpen, this->_boldSymbolClose); });
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/TextEdit.cpp]: Inside lambda body, you shouldn't use this. It overloads code.

File: Client/Client.Qt/Widgets/TextEdit.cpp (hunk 28 above)
Comment (Orange-hanter) [reply Client/Client.Qt/Widgets/TextEdit.cpp:99]: Put response here, please

File: Client/Client.Qt/Widgets/TextEdit.hpp [hunk 32]
```diff
@@ -47,16 +47,16 @@ public slots:
...
+    void delSymbolsOutSelection(QString& text, int start, int end, int symbolSize) const;
+    void insertSymbolsInSelection(QTextCursor& cursor, int start, int end, int symbolSize, const QString& symbolStart,
+                                  const QString& symbolEnd) const;
+    void selectText(QTextCursor& cursor, int start, int end) const;
 
-private:
+// variable field
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/TextEdit.hpp]: Remove please this comment. It useless

File: Client/Client.Qt/Widgets/TitleWidget.cpp [hunk 33]
```diff
@@ -22,20 +22,20 @@ TitleWidget::TitleWidget(QWidget* parent, const Style::TitleBar& st) : QWidget(p
...
+    _closeButton->setClickCallback([parent, this]() {
         parent->deleteLater();
-        _closeButton->clearState();
+        this->_closeButton->clearState();
     });
-    _minimizeButton->setClickCallback([=]() {
+    _minimizeButton->setClickCallback([parent, this]() {
         parent->showMinimized();
-        _minimizeButton->clearState();
+        this->_minimizeButton->clearState();
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/TitleWidget.cpp]: This inside lambda body overloads code

**Review by memorisecodead — COMMENTED** [review]
What do you have left to do?

File: Client/Client.Qt/Widgets/LoadingSpinner.hpp (hunk 24 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/LoadingSpinner.hpp]: @Orange-hanter , @ivanboika .I did

File: Client/Client.Qt/Widgets/Menu.hpp (hunk 25 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/Menu.hpp:86]: @Orange-hanter , @ivanboika .I did

File: Client/Client.Qt/Widgets/MessageWidget.cpp (hunk 26 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/MessageWidget.cpp:14]: @Orange-hanter , @ivanboika .I did

File: Client/Client.Qt/Widgets/TextEdit.cpp (hunk 31 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/TextEdit.cpp]: @Orange-hanter , @ivanboika .I did

File: Client/Client.Qt/Widgets/TitleWidget.cpp (hunk 33 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/TitleWidget.cpp]: @Orange-hanter , @ivanboika .I did

File: Client/Client.Qt/Widgets/TextEdit.cpp (hunk 28 above)
Comment (Gorba41kD) [reply Client/Client.Qt/Widgets/TextEdit.cpp:99]: @Orange-hanter , @ivanboika . We discussed this at the meeting. You closed this issue.

File: Client/Client.Qt/Widgets/TitleWidget.cpp (hunk 33 above)
Comment (memorisecodead) [reply Client/Client.Qt/Widgets/TitleWidget.cpp]: now, it looks good, we specify what objects we work with and how we transfer them, depending on the task

File: Client/Client.Qt/Style/StyleWidgets.hpp [hunk 34]
```diff
@@ -18,12 +18,12 @@ struct ScrollArea
...
-    /// deltab: pixels
-    int deltab;
-    /// deltax: pixels
-    int deltax;
+    /// deltaT: pixels
+    int deltaT;
+    /// deltaB: pixels
+    int deltaB;
+    /// deltaX: pixels
+    int deltaX;
```
Comment (Orange-hanter) [inline Client/Client.Qt/Style/StyleWidgets.hpp]: Obvious comments

Comment (Gorba41kD) [reply Client/Client.Qt/Style/StyleWidgets.hpp]: @Orange-hanter ,  I agree with you on that. That's why I removed these useless comments in this header

**Review by Orange-hanter — APPROVED** [review]

**Comment by Orange-hanter** [comment]
Before pulling to dev branch, manually test all features

File: Client/Client.Qt/Style/StyleWidgets.hpp [hunk 35]
```diff
@@ -11,256 +11,181 @@ namespace internal
 /// StyleWidgets initializer
 void init_StyleWidgets(int scale);
 
-}  // namespace internal
+}  /// namespace internal
```
Comment (memorisecodead) [inline Client/Client.Qt/Style/StyleWidgets.hpp:14]: remove one "/", it is not needed there.

File: Client/Client.Qt/Style/StyleWidgets.hpp [hunk 36]
```diff
@@ -11,256 +11,181 @@ namespace internal
...
     Style::TitleBarButton closeButton;
-    /// maximizeButton: Style::TitleBarButton
     Style::TitleBarButton maximizeButton;
-    /// restoreButton: Style::TitleBarButton
     Style::TitleBarButton restoreButton;
-    /// minimizeButton: Style::TitleBarButton
     Style::TitleBarButton minimizeButton;
 };
 
 /// Style struct for Menu
```
Comment (memorisecodead) [inline Client/Client.Qt/Style/StyleWidgets.hpp:126]: Why not tweak the documentation for the current classes?

File: Client/Client.Qt/Style/StyleWidgets.hpp [hunk 37]
```diff
@@ -11,256 +11,181 @@ namespace internal
...
+    int    innerRadius;
     double minimumTrailOpacity;
-    /// trailFadePercentage: double
     double trailFadePercentage;
 };
 
-}  // namespace Style
+}  /// namespace Style
 
 namespace st
```
Comment (memorisecodead) [inline Client/Client.Qt/Style/StyleWidgets.hpp:164]: What you mean about namespace "st"?

File: Client/Client.Qt/Widgets/MessageWidget.cpp [hunk 38]
```diff
@@ -61,7 +50,7 @@ void MessageWidget::paintEvent(QPaintEvent* e)
     p.setFont(getStyle().fontname);
     if (_messageFlags & MessageFlag::Deleted)
     {
-        QString dltmsg = QString("Message was deleted");
+        auto dltmsg = QString("Message was deleted");
```
Comment (memorisecodead) [inline Client/Client.Qt/Widgets/MessageWidget.cpp:53]: Why you use "auto" when we knowing about type of statement?

File: Client/Client.Qt/Widgets/MessageWidget.cpp [hunk 39]
```diff
@@ -90,7 +79,7 @@ void MessageWidget::resizeEvent(QResizeEvent* e)
     int replyButtonX = width() - getStyle().radius - _menuBtn->width() - _replyBtn->width() - 2;
     _replyBtn->move(replyButtonX, getStyle().radius);
     getFmtMessageText()->resize(width() - getStyle().radius * 4 - 1, getFmtMessageText()->document()->size().height());
-    _reactions->move(getStyle().radius * 2, getFmtMessageText()->y() + getFmtMessageText()->document()->size().height() + getStyle().radius);
+    _reactions->move(getStyle().radius * 2, static_cast<int>(getFmtMessageText()->y() + getFmtMessageText()->document()->size().height() + getStyle().radius));
```
Comment (memorisecodead) [inline Client/Client.Qt/Widgets/MessageWidget.cpp:82]: What changed after rounding, is it really necessary?

File: Client/Client.Qt/Widgets/MessageWidget.cpp [hunk 40]
```diff
@@ -157,7 +146,7 @@ int MessageWidget::expectedHeight() const
     {
         return getStyle().fontname->height + getStyle().radius * 2;
     }
-    return getStyle().radius * 7 + getStyle().fontname->height + getFmtMessageText()->document()->size().height() + _reactions->layoutSize().height();
+    return static_cast<int>(getStyle().radius * 7 + getStyle().fontname->height + getFmtMessageText()->document()->size().height() + _reactions->layoutSize().height());
```
Comment (memorisecodead) [inline Client/Client.Qt/Widgets/MessageWidget.cpp:149]: Also there

**Pull request closed without merging**